sky/skylet/log_lib.py

```diff
@@ -377,7 +379,10 @@ def tail_logs(job_owner: str,
     log_path = os.path.join(log_dir, 'run.log')
     log_path = os.path.expanduser(log_path)
 
-    status = job_lib.get_status(job_id)
+    job_lib.update_job_status(job_owner, [job_id], need_output=False)
+    # No need to lock the status here, as the while loop can handle the
+    # older status.
+    status = job_lib.get_status_no_lock(job_id)
```

Comment:
How about making update_job_status() return the values that just got set inside? Same for L408.